torch/csrc/autograd/engine.cpp

```diff
@@ -428,11 +428,14 @@ void validate_outputs(
     ss << edges.size() << ", but got " << grads.size();
     AT_ERROR(format_error(ss.str()));
   }
+
   for (size_t i = 0; i < grads.size(); i++) {
     const auto& edge = edges[i];
     if (!edge.is_valid()) continue;
 
     const auto& metadata = edge.function->input_metadata(edge.input_nr);
+    // If any input grad is lazy, we don't check shape properties
```

Comment:
This comment is missing the most important part: the part where you say it's because of the HACK where we don't have accurate dtype or shape information on lazy tensors. In fact, there should probably be a pretty big Note somewhere about this state of affairs, cross-linked from here.

Reply:
Still not fixed (there's a comment about this in the design doc in LazyTensor.cpp but you need a cross reference here)

Reply:
Here's how I like to think about whether or not a cross reference is necessary: suppose that I'm a random developer who hasn't seen LazyTensor.cpp, and happens to be reading this code. Is it clear where I would go to find out more information about this weird line of code? If it's not obvious, I need a cross reference.